### PR TITLE
objecter: fix linger cancel link bug 10828

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
actually not bug 10828; removed from commit msg.  i don't see an open bug.

File: src/osdc/Objecter.cc [hunk 1]
```diff
@@ -3007,13 +3015,15 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
   if (op->oncommit_sync) {
```
Comment (jdurgin) [inline src/osdc/Objecter.cc]: should this have && (m->is_ondisk() || rc) like oncommit as well?

**Comment by jdurgin** [comment]
one related comment, though this change looks good.
assuming it's been tested, Reviewed-by:

File: src/osdc/Objecter.cc (hunk 1 above)
Comment (liewegas) [inline src/osdc/Objecter.cc]: may as well, although in reality we never set the callback on a non-commit request.

**Comment by liewegas** [comment]
@jdurgin refactored that piece, look ok?

**Comment by jdurgin** [comment]
Looks good to me

**Pull request merged**